### ISSUE TITLE
[Bug] Core changes to ImmutableOpenMap causing various failures 

### ISSUE THREAD

**Issue by scrawfor99** [opensearch-project/security#2714]
Builds against main (and potentially other versions) are encountering issues related to the `valuesIt()` method used by the `clusterService` when it tries to look at the list of indices. 

```bash
/home/runner/work/security/security/src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java:652: error: cannot find symbol
                   return clusterService.state().getMetadata().getIndices().valuesIt();
```
These changes are the result of the Core refactor to no longer use `ImmutableOpenMap`. 

_Originally posted by @willyborankin in https://github.com/opensearch-project/security/issues/2557#issuecomment-1517988828_

**Comment by scrawfor99** [comment]
I expect this change caused the issue: https://github.com/opensearch-project/OpenSearch/commit/1ab22e1cb3d52101b72741447f7d68333b43cc0a

Working on swapping over now.

**Comment by scrawfor99** [comment]
[Triage] @scrawfor99 is working on a fix for this issue at #2715.

**Comment by DarshitChanpura** [comment]
Closing this as related PRs have been merged.
